modules/core/arrow-typeclasses/src/main/kotlin/arrow/typeclasses/MonadError.kt

```diff
@@ -131,4 +131,14 @@ interface MonadThrow<F> : MonadError<F, Throwable> {
     wrapReturn.startCoroutine(continuation, continuation)
     return continuation.returnedMonad()
   }
+
+  override fun <B> binding(c: suspend MonadContinuation<F, *>.() -> B): Kind<F, B> =
+    bindingCatch { c() }
+
+  override fun <A> fx(f: suspend MonadContinuation<F, *>.() -> A): Kind<F, A> =
+    bindingCatch { f() }
+
+  fun <A> fx(f: suspend MonadErrorContinuation<F, *>.() -> A, unit: Unit = Unit): Kind<F, A> =
```

Comment:
How do you desambiguate them later? when the user does their first raiseError inside the block?

Reply:
With the multiple Fx instances. Each monad targets a single `Fx` instance with the right powers. That allows us to have for types like `IO` only the `Concurrent<F>` `fx` version available.